packages/cli/src/server/http-server.ts

```diff
@@ -17,7 +26,54 @@ export class HttpServer {
     });
   }
 
+  public redirectResponse(): Promise<string> {
+    return new Promise((resolve) => {
+      const interval = setInterval(() => {
+        if (this.token) {
+          clearInterval(interval);
+          resolve(this.token);
+        }
+      }, 300);
+    });
+  }
+
   public close(): void {
     this.server.close();
   }
+
+  private handleRedirectRequest(req: http.IncomingMessage) {
+    this.token = new URLSearchParams(req.url.slice(REDIRECT_ROUTE.length)).get('token');
+  }
+
+  private async handleWidgetDemo(res: http.ServerResponse): Promise<void> {
+    return new Promise((resolve) => {
+      const dashboardPath = path.resolve(__dirname, '../constants/dashboard/index.html');
+
+      fs.readFile(dashboardPath, 'utf8', (error, content) => {
+        if (error) {
+          if (error.code === 'ENOENT') {
+            res.end('404', 'utf-8');
+          } else {
+            res.end('500');
+          }
+        } else {
+          const payLoad = JSON.parse(this.config.getValue('triggerPayload'));
+
+          if (!payLoad) {
+            res.end('500');
+          }
+
+          payLoad.forEach((param) => {
+            // eslint-disable-next-line no-param-reassign
+            content = content.replace(`REPLACE_WITH_${param.key}`, param.value);
+          });
+
+          res.writeHead(200);
+          res.end(content, 'utf-8');
+        }
+      });
+
+      resolve();
```

Comment:
This resolve might happen before the fs.read finishes, Is that the expected behaviour?

Reply:
I do not expect it to cause any problems but this was ultimately not the intention here. Thanks for noticing I'll post a fix soon.